src/components/HyperlinkingWrapper.js

```diff
@@ -77,6 +77,14 @@ export default class HyperlinkingWrapper extends Component {
 
                 this.setState({ currentSelection });
                 this.quill.format('highlight', true);
+            } else {
+                const [blot, blotRange] = this.getBlotFromIndex(range.index);
+                const format = blot.formats() || {};
+                const hasFormatting = Object.keys(format).length > 0;
```

Comment:
can we check if the selected blot is link blot using `blot.statics.blotName` ? Then we could get rid of `hasFormatting`.

Reply:
done